### ISSUE TITLE
java.lang.NullPointerException at android.content.ContextWrapper.getApplicationInfo(ContextWrapper.java:148)

### ISSUE THREAD

**Issue by rockerhieu** [robolectric/robolectric#1661]
I'm getting a weird NPE when trying `3.0-rc2`:

```
java.lang.NullPointerException
	at android.content.ContextWrapper.getApplicationInfo(ContextWrapper.java:148)
	at com.xxx.DownloadServiceTest.testDownloadFailure(DownloadServiceTest.java:132)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:298)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:292)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:745)
```

Here is the code:
```java
 @Test(timeout = DEFAULT_TIMEOUT)
  public void testDownloadFailure() throws Exception {
    doAnswer(new Answer() {
      @Override
      public Object answer(InvocationOnMock invocationOnMock) throws Throwable {
        AsyncCallback<Uri> asyncCallback = (AsyncCallback<Uri>) invocationOnMock.getArguments()[4];
        asyncCallback.onFailure(new IOException());
        return null;
      }
    }).when(mChatController) .downloadFile(any(String.class), any(String.class),
        any(String.class), any(DownloadProgressListener.class), any(AsyncCallback.class));
    mDownloadServiceSpy.onHandleIntent(createServiceIntent());
    verify(mDownloadServiceSpy).onFailure(mThrowableArgumentCaptor.capture());
    assertThat(mThrowableArgumentCaptor.getValue(), instanceOf(IOException.class));
  }

  private Intent createServiceIntent() {
    Intent intent = new Intent(RuntimeEnvironment.application.getBaseContext(), UpdateAppService.class);
    intent.putExtra(DownloadService.EXTRA_FILE_NAME, RuntimeEnvironment.application.getString(R.string.app_name) + "_1");
    intent.putExtra(DownloadService.EXTRA_NOTIFICATION_TITLE, RuntimeEnvironment.application.getString(R.string.app_name));
    intent.putExtra(UpdateAppService.EXTRA_VERSION_CODE, 1);
    intent.setData(Uri.parse("http://xxx.com"));
    return intent;
  }
```

It was running fine with `2.4`. If you need any other information, please let me know. Thanks!

**Comment by jongerrish** [comment]
Why are you constructing your intent with

RuntimeEnvironment.application.getBaseContext()?

Why not just

RuntimeEnvironment.application?

**Comment by erd** [comment]
This should be fixed in 5b4704fe816250e9643505b84c7a5a89b7979e0d.
